game/script-topics.rpy

```diff
@@ -816,6 +816,280 @@ label talk_eating_well:
+                    n "Of course if you insist on retyping it I won't stop you :p"
+                    n "Alright so one sec I'll just make a quick form for you to type in"
+                    $ renpy.pause(3.0, hard=True)
+                    n "Alright here it comes!"
+                    n "Just save and close it when you're done"
+                    while not persistent.weather_api_key:
+                        $ persistent.weather_api_key = txt_input("API key here: ")
+                        if not persistent.weather_api_key:
+                            n "Hey! I see what you're doing!"
+                            n "Let's try this again and do it properly this time"
+                    n "Thanks a bunch!"
+                    n "Seems like it might take a while to activate"
+                    n "I'll tell you once something changes"
+                    $ weather.set_next_weather_call_time(7920)
+
+        "sorry, I'm busy right now":
+            n "oh...{w=0.3} Well...{w=0.1} that's alright"
+            #TODO: unlock a topic for seting this up later
+    return {"lock" : None}
+
+init 5 python:
+    registerTopic(
+        Topic(
+            persistent._topic_database,
+            label="talk_weather_setup_part2",
+            unlocked=False,
+            nat_says=True,
+            category=["weather"]
+        ),
+        topic_group=TOPIC_TYPE_NORMAL
+    )
+
+label talk_weather_setup_part2:
+    if not weather.is_api_key_valid(persistent.weather_api_key):
+        n "Hey about that weather thing"
+        n "It's been like 2 hours and still nothing's happened"
+        n "You sure you gave me the right thing?"
+        # Why the heck doesn't renpy have a for loop
+        # reeeeeeee
+        $ i = 0
+        while i<5:
+            $ i+=1
+            n "Let's try this again"
+            $ persistent.weather_api_key = txt_input("API key: ")
+            if not weather.is_api_key_valid(persistent.weather_api_key):
+                #TODO: help player troubleshoot this
+                n "Still nothing..."
+            else:
+                jump talk_weather_setup_part2
+        # If it doesn't work at this point it's probably best to give up
+        n "Aww man nothing seems to work"
+        n "Whatever, let's try it later"
+        #TODO: randomly bring it up again
+    else:
+        n "Hey hey! The weather thing is finally doing something!"
+        n "Hmmm it says 'nothing to geocode'"
+        n "Ah, it has no idea where to look for weather"
+        n "I was so scared to ask you this that I completely forgot about it"
+        n "In order to share weather with you...{w=0.3} I need to know your location"
+        n "N-Not that I couldn't find it myself! {w=0.3}I just thought I should probably ask you."
+        n "So...{w=0.2} do you mind if I know where you live?"
+        menu:
+            "So... do you mind if I know where you live?"
+
+            "No big deal. I trust you Nat.":
```

Comment:
I think until we gain nickname functionality, we should try to keep things consistent by using Natsuki over any shortened version in any dialogue seen by the player (though using "Nat" in comments is fine, imo)!